### DISCUSSION TITLE
Having native aarch64 images to to run on aarch64 hosts

### DISCUSSION THREAD

**Discussion by BlackDex** [BlackDex/rust-musl#8] [Ideas]
As requested here #4.
It might be an idea to build these images.
That does mean i have to create more toolchain images which use a aarch64 host compiler.
And, also create new musl-base images either via some free hosted aarch64 runners or via QEMU, which will probably take a while for those to be build.

**Comment by Azathothas** [comment]
Hi, @BlackDex 

- [Google Cloud](https://cloud.google.com/compute/docs/instances/arm-on-compute) Offers/offered free credits for `Tau T2A VMs` until `March 31 2024`.
> Ref: https://cloud.google.com/compute/docs/instances/arm-on-compute
> Note: IPv4 & Storage will still be charged, but this comes to about 1-2$ for me, I have been using a `4vCPU 16GB RAM`  build machine for arm64 using Self-Hosted Runners: https://github.com/Azathothas/Toolpacks/actions/workflows/build_aarch64_Linux.yaml

- AWS also offers free trial for `EC2 t4g.small` `until Dec 31st 2024` 
> Ref: https://aws.amazon.com/ec2/instance-types/t4/
>      https://aws.amazon.com/ec2/faqs/#t4g-instances
> Note: Unlike Google, AWS includes free IPv4  & Storage (Standard 30 GB) for free.

And lastly, buildx is itself capable of [cross-building](https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/) docker images. I see you already use buildx so you might already be aware.
See: https://docs.docker.com/build/building/multi-platform/
Here's an example workflow, I use this to build self-hosted runner images for arm64: [build_gh_runner_images.yaml](https://github.com/Azathothas/Toolpacks/blob/main/.github/workflows/build_gh_runner_images.yaml)

**Reply by BlackDex** [reply]
@Azathothas I do not know if you are still using my images. But i have created aarch64 images since just now.
You can just pull the same image tags on a aarch64 host.

And, if you want, you can also verify the images using GitHub attestation.
